.github/workflows/iroha2-dev-pr-wasm.yaml

```diff
@@ -41,20 +49,26 @@ jobs:
           retention-days: 1
 
   tests:
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
     needs: build_executor
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-04-18
     steps:
       - uses: actions/checkout@v4
-      - uses: Swatinem/rust-cache@v2
-      - name: Download executor.wasm file
+      - name: Download executor.wasm
         uses: actions/download-artifact@v4
         with:
           name: executor.wasm
           path: ${{ env.DOCKER_COMPOSE_PATH }}
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          targets: aarch64-apple-darwin,wasm32-unknown-unknown
+          # needed to build smart contracts
+          components: rust-src
+      - uses: Leafwing-Studios/cargo-cache@v2
+        with:
+          sweep-cache: true
       - name: Install iroha_wasm_test_runner
         run: cargo install --path tools/wasm_test_runner
       - name: Run smart contract tests on WebAssembly VM
         working-directory: smart_contract
-        run: mold --run cargo test -p iroha_smart_contract -p iroha_smart_contract_utils --release --tests --target wasm32-unknown-unknown --no-fail-fast --quiet
```

Comment:
Is it better without mold?

Reply:
There's no mold on mac (except a commercial version), and mac's linker is as good as mold